packages/core/src/__tests__/dispatcher.test.ts

```diff
@@ -35,6 +35,16 @@ const ipfs = {
   id: async () => ({ id: 'ipfsid' }),
 }
 
+const recordCache = {
```

Comment:
ah, I see, this is why you were passing in the recordCache rather than constructing it within the dispatcher.  For one thing, you can actually modify the contents of an object out from under it in javascript, so you could still have provided a mock record cache to the dispatcher even while having the dispatcher construct it itself.

But also, I don't understand why you're providing a fake recordCache at all?  It doesn't do any networking or I/O, so it shouldn't be hard to use the real thing, and mocking this out just weakens the value of the test IMO.  For instance, I'm wondering if you used the real cache if the tests would have caught the issue I mentioned about CID object equality not working in the cache (or at least, it *could* repro the issue if the test used multiple CID instances for the same CID)

Reply:
The modified test and the new test were mainly to test the cache hit and miss code paths. The problem with using the real thing is that I won't be able to check for a cache miss and force a lookup in IPFS 😕
 
I guess I could look for a CID I haven't stored yet and assume both calls will be performed but will return null? 

I was also going to add one more test to check whether eviction of the oldest entry caused IPFS to lookup the entry but that's probably unnecessary since that would just be testing `LRUMap`.

Reply:
Actually, I won't be able to check the two code paths without a mocked cache.

Reply:
hmm okay, I don't totally follow but if you think it's important to test with a mocked cache that's fine by me.  Maybe we could *also* have a small test that uses the real thing?  That could also make sure that 2 different CID instances for the same data are handled properly by the cache

Reply:
Took a bit to wrap my head around your suggestion. I'm now testing with just the real cache and able to test both the cache hit/miss legs like I wanted.